### PR TITLE
Reserved words handling + Support for data store selection

### PR DESCRIPTION
**Pull request by jacknagz**
## Summary
- Refactor reserved words implementation to follow official Snowflake documentation precisely
- Add support for quoting reserved words inside function calls (e.g., `COUNT(DISTINCT account)`)
- Implement smart filtering to avoid inappropriately quoting SQL keywords
- Add comprehensive test coverage including real-world VPC Flow logs scenarios

## Test plan
- [x] All existing tests pass (105 passed, 4 skipped)
- [x] New comprehensive test suite covers success and error cases
- [x] Real-world AWS VPC Flow logs query example tested and working
- [x] Function call detection prevents quoting function names
- [x] CASE WHEN expressions work correctly
- [x] Forbidden keyword validation works as expected
- [x] Linting passes

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/mcp_panther/panther_mcp_core/tools/data_lake.py [hunk 1]
```diff
@@ -145,6 +370,15 @@ async def execute_data_lake_query(
     """
     logger.info("Executing data lake query")
 
+    # Validate and wrap reserved words according to Snowflake constraints
+    sql, validation_error = _validate_and_wrap_reserved_words(sql)
+    if validation_error:
+        logger.error(validation_error)
+        return {
+            "success": False,
+            "message": validation_error,
```
Comment (darwayne) [inline src/mcp_panther/panther_mcp_core/tools/data_lake.py:379]: what are your thoughts on including the reserved words in the failure response as well so llm won't make same mistake if they had multiple words that were reserved?

**Review by darwayne — COMMENTED** [review]
Looks mostly good just had a question

File: src/mcp_panther/panther_mcp_core/tools/data_lake.py (hunk 1 above)
Comment (darwayne) [reply src/mcp_panther/panther_mcp_core/tools/data_lake.py:379]: yeah. Doesn't seem like a ton to add to the context.

**Review by darwayne — APPROVED** [review]

**Comment by jacknagz** [comment]
## 🚀 Additional Updates Since Initial PR

This PR has been significantly enhanced with additional improvements beyond the original data lake query management functionality:

### 📊 **New Data Lake Query Management Tools**
- **`list_data_lake_queries()`** - Comprehensive query monitoring with filtering by status, date ranges, content search, and pagination
- **`cancel_data_lake_query()`** - Resource management tool to terminate long-running queries and prevent system overload

### 🏗️ **Enhanced Type Safety & Modern Annotations**
- Migrated from `Optional[T]` to modern `T | None` syntax following fastmcp best practices
- Added `QueryStatus(str, Enum)` for type-safe status filtering
- Implemented proper MCP tool annotations:
  - `readOnlyHint: true` for monitoring operations
  - `destructiveHint: true` for query cancellation
- Added regex patterns for ISO 8601 datetime validation

### 🔍 **Comprehensive Test Coverage**
- Added 9 new test functions covering success, error, and edge cases
- Tests for both user-issued and API token-issued queries
- Validation of filtering, pagination, and error handling
- All tests passing with improved type safety

### 📋 **GraphQL Schema Integration**
- Added `LIST_DATA_LAKE_QUERIES` query with complete field selection
- Added `CANCEL_DATA_LAKE_QUERY` mutation for resource management
- Proper handling of user vs API token issuer information

### ⚡ **Key Benefits**
- **Performance Monitoring**: Identify and cancel resource-intensive queries
- **System Health**: Prevent data lake overload during peak usage
- **Operational Visibility**: Track query history, timing, and ownership
- **Type Safety**: Enum-based validation prevents runtime errors
- **User Experience**: Clear error messages and comprehensive filtering

### 🎯 **Common Use Cases**
```python
# Find running queries that need attention
list_data_lake_queries(status=[QueryStatus.RUNNING])

# Monitor recent query activity
list_data_lake_queries(started_at_after="2024-01-01T00:00:00.000Z")

# Cancel problematic queries
cancel_data_lake_query("long-running-query-id")
```

The implementation follows all fastmcp best practices and provides essential tools for managing data lake performance and preventing system overload. 🎉
